packages/material-ui/src/AppBar/AppBar.d.ts

```diff
@@ -17,6 +17,4 @@ export type AppBarClassKey =
   | 'colorPrimary'
   | 'colorSecondary';
 
-declare const AppBar: React.ComponentType<AppBarProps>;
```

Comment:
@eps1lon @oliviertassinari Would you rather I didn't change how the component is declared?

Reply:
The easiest thing would be to change them to be consistent, as currently, they aren't.

Reply:
I would even prefer not using the types from `@types/react` so that we have more control over the structure. For example we don't need to use a union here (problems with mapped types). A plain function declaration with the props should work and also allows a generic component type.

Reply:
So this is fine with you?
```typescript
export default function ComponentName(props: ComponentNameProps): JSX.Element;
```